beancount.el

```diff
@@ -490,12 +490,14 @@ With an argument move to the next non cleared transaction."
          ;; that begin with something that looks like a
```

Comment:
I don't understand what this improves. If you don't want completion while editing a string, don't invoke the completion function. What am I missing?

Reply:
The problem with "don't invoke the completion function" is that we use a tab-dwim function which either toggles outline sections, attempts to complete, or indents based on context.

My goal is to be able to hit TAB to indent while editing in strings regardless of their content (in particular if the string happens to have `#` or `^` in it), and IMO there's no case where a user would want `beancount-mode` to try to complete anything in a string (with the exception of options), since strings are more-or-less opaque to beancount.

Reply:
I see where this is coming from now: you enabled completion of tags and links also now when on a dedicated line. Very welcome fix BTW, thank you! This enables completion of the sames in string literals too. 

I admit I see advantages in permitting completion of tag and links in strings too. It is common to mention links or tags in strings for many different reasons, and account names too. Having auto completion for them in strings could be seen as an handy feature. If you don't want completion just put hit space or close the double quotes before pressing tab. I am almost tempted to open up account completion to more contexts rather to restrict existing completion contexts further.

Reply:
I don't feel strongly about not completing in strings to fight you too hard here, but I don't personally see a need for any kind of completion in strings in Beancount. I could see it being more useful in comments than in strings.

I do, however, feel strongly that `beancount-collect` should not grab matches that occur in strings (at least not in the generic case, perhaps there may be some exceptions in the future). In fact, I think that it should probably also ignore matches that occur in comments.